nautobot/docs/development/jobs/job-patterns.md

```diff
@@ -0,0 +1,519 @@
+
+<!-- pyml disable-num-lines 10 proper-names -->
+!!! example
+    ```py
+    username = self.user.username
+    job_result_id = self.job_result.id
+    self.logger.info("Job %s initiated by user %s is running.", job_result_id, username)
+    ```
+
+You can access other standard Django user attributes like `self.user.email`, `self.user.is_staff`, or use `self.user.has_perm(...)` for permission checks.
+
+### Logging Patterns
+
++/- 2.0.0
+
+Nautobot logs messages from Jobs in a structured way, storing them as part of the `JobResult` and displaying them in the UI. This enables Jobs to provide real-time feedback, track their progress, and surface success or failure messages clearly to the user.
+
+Use the `self.logger` property to write log messages from within your Job code. These logs appear in the UI and are also saved as [`JobLogEntry`](../../user-guide/platform-functionality/jobs/models.md#job-log-entry) records associated with the current [`JobResult`](../../user-guide/platform-functionality/jobs/models.md#job-results).
+
+#### Logging Levels
+
+Nautobot supports standard logging levels, with additional custom levels for success and failure messages:
+
+| Level   | Method                 | Description                               |
+|---------|------------------------|-------------------------------------------|
+| DEBUG   | `self.logger.debug()`  | Detailed diagnostic information.          |
+| INFO    | `self.logger.info()`   | General operational messages.             |
+| SUCCESS | `self.logger.success()`| Indicates successful operations.          |
+| WARNING | `self.logger.warning()`| Signals potential issues.                 |
+| FAILURE | `self.logger.failure()`| Denotes failed operations.                |
+| ERROR   | `self.logger.error()`  | Serious errors that prevent execution.    |
+| CRITICAL| `self.logger.critical()`| Critical conditions.                      |
+
+!!! note
+    `logger.success()` and `logger.failure()` were introduced in versions 2.4.0 and 2.4.5, respectively.
+
+#### Writing Log Messages
+
+<!-- pyml disable-num-lines 10 proper-names -->
+!!! example
+    ```py
+    self.logger.info("Job is starting.")
+    self.logger.error("An unexpected error occurred.")
+    self.logger.success("Provisioning complete.")
+    ```
+
+For most use cases, you'll use `self.logger`. You can also obtain the same logger via `nautobot.extras.jobs.get_task_logger(__name__)`, though this is less common.
+
+#### Structured Log Context with `extra`
+
+You can attach structured metadata to log messages using the `extra` parameter. This enables grouping and improves how logs are displayed or queried:
+
+- `grouping`: A logical group name for related messages, shown in the UI as expandable sections
+- `object`: A Nautobot object instance to associate with this log message (e.g., a Device)
+- `skip_db_logging`: Set to `True` to avoid saving the log message in the database
```

Comment:
```suggestion
- `skip_db_logging`: Set to `True` to avoid saving the log message in the database (it will still be visible in the Celery worker log output)
```